streams/jvm/src/main/scala/zio/stream/platform.scala

```diff
@@ -484,6 +484,34 @@ trait ZStreamPlatformSpecificConstructors {
               }
     } yield conn
 
+  /**
+   * Create a Connection from client socket
+   */
+  def fromSocketClient(
+    port: Int,
+    host: String
+  ): ZIO[Blocking, Throwable, UManaged[Connection]] =
```

Comment:
This is a great addition @searler, thank you for contributing.

Since this is not an actual streaming operation (but rather the operations on `Connection` itself are), I am hesitant to place this here.

This is probably because we should create a separate module for these I/O operations, as we do with the `compression` operations.

What do you think @jdegoes @adamgfraser?